dcs-packaging-tool/dcs-packaging-tool-gui/src/main/java/org/dataconservancy/packaging/gui/presenter/impl/PackageMetadataPresenterImpl.java

```diff
@@ -0,0 +1,222 @@
+                        }
+                    } else if (node instanceof DatePicker) {
+                        LocalDate localDate = ((DatePicker) node).getConverter().fromString(getController().getPackageState().getPackageMetadataValues(node.getId()).get(0));
+                        ((DatePicker) node).setValue(localDate);
+                    }
+                }
+            }
+        }
+    }
+
+    private void bind() {
+
+        // FIXME: The profile names should come from an actual service
+        view.loadDomainProfileNames(Arrays.asList("Bag-It", "Custom Profile", "Custom Profile 2"));
+        view.getAddDomainProfileButton().setOnAction(event -> {
+            if (view.getDomainProfilesComboBox().getSelectionModel().getSelectedItem() != null &&
+                    !view.getDomainProfilesComboBox().getSelectionModel().getSelectedItem().isEmpty()) {
+                view.addDomainProfileRemovableLabel(view.getDomainProfilesComboBox().getSelectionModel().getSelectedItem());
+            }
+        });
+
+        if (!view.isFormAlreadyDrawn()) {
+            view.setupRequiredFields(packageMetadataService.getRequiredPackageMetadata());
+            view.setupRecommendedFields(packageMetadataService.getRecommendedPackageMetadata());
+            view.setupOptionalFields(packageMetadataService.getOptionalPackageMetadata());
+        }
+
+        view.getContinueButton().setOnAction(event -> {
+            updatePackageState();
+
+            if (validateRequiredFields()) {
+                view.getErrorLabel().setVisible(false);
+                getController().goToNextPage();
+            } else {
+                view.getErrorLabel().setText(errors.get(ErrorKey.MISSING_REQUIRED_FIELDS));
+                view.getErrorLabel().setVisible(true);
+            }
+
+        });
+
+        view.getSaveButton().setOnAction(event -> {
+            updatePackageState();
+            packageMetadataFile = controller.showSaveFileDialog(view.getPackageMetadataFileChooser());
+            // TODO: Store the package metadata in some sort of file here.
+        });
+
+        view.getCancelLink().setOnAction(event -> view.showWarningPopup());
+
+        if (Platform.isFxApplicationThread()) {
+            view.getWarningPopup().setCancelEventHandler(event -> view.getWarningPopup().hide());
+            view.getWarningPopup().setConfirmEventHandler(event -> {
+                view.getWarningPopup().hide();
+                getController().goToPreviousPage();
+            });
+        }
+
+    }
+
+    /**
+     * helper method that updates the PackageState in the controller with the values in the form.
+     */
+    private void updatePackageState() {
+
+        getController().getPackageState().setPackageName(view.getPackageNameField().getText());
+
+        // Clear the package metadata list and reset the values based on the current state of form
+        getController().getPackageState().setPackageMetadataList(new LinkedHashMap<>());
```

Comment:
This looks like it would clear everything that's in this list. This list would eventually contain system generated metadata. But that's not being put in until the last step. Except for maybe BAGIT PROFILE ID.